controller/transfer/util.py

```diff
@@ -141,6 +145,19 @@ def check_and_convert_category_for_unknown(
         )
 
 
+def check_and_covert_nested_attributes_to_text(df: pd.DataFrame) -> pd.DataFrame:
+    for key in df.columns:
+        if [value for value in pick_sample(df, 10, key) if type(value) == dict]:
+            df[key] = df[key].apply(lambda x: json.dumps(x))
+    return df
+
+
+def pick_sample(df: pd.DataFrame, sample_size: int, key: str) -> pd.DataFrame:
+    if len(df[key]) <= sample_size:
```

Comment:
Suggestion (not sure if it's a valid one :D). :
We can move the `sample_size` parameter as the last one, and set it default to 10, so we don't have to pass it every time we want to use the function (if we want to)

Reply:
Good suggestion I will include it :)